lib/array.rb

```diff
@@ -0,0 +1,83 @@
+# class Array
+#   def my_inject(arg = nil, &block)
+#     if arg.nil?
+#       my_inject_without_arg &block
+#     else
+#       if arg.is_a? Symbol
+#         my_inject_with_symbol symbol, &block
+#       else
+#         my_inject_with_arg arg, &block
+#       end
+#     end
+#   end
+#
+#   def my_inject_with_arg arg, &block
+#     result = arg
+#     each do |value|
+#       result = yield(result, value)
+#     end
+#     result
+#   end
+#
+#   def my_inject_without_arg &block
+#     result = self.first
+#     self[1..-1].each do |value|
+#       result = yield(result, value)
+#     end
+#     result
+#   end
+#
+#   def my_inject_with_symbol symbol, &block
+#     # case symbol
+#   end
+# end
+
+class Array
+  def my_inject(arg = nil, arg_sym = nil, &block)
+    if arg.nil?
+      my_inject_without_arg(&block)
+    else
+      if arg.is_a?(Symbol)
+        my_inject_with_symbol(arg, &block)
+      else
+        if arg && arg_sym
+          my_inject_with_arg_and_symbol(arg, arg_sym, &block)
+        else
+          my_inject_with_arg(arg, &block)
+        end
+      end
+    end
+  end
+
+  def my_inject_with_arg(arg)
+    result = arg
+    each do |value|
+      result = yield(result, value)
+    end
+    result
+  end
+
+  def my_inject_without_arg
+    result = self.first
+    self[1..-1].each do |value|
+      result = yield(result, value)
+    end
+    result
+  end
+
+  def my_inject_with_symbol(arg_is_sym, &block)
```

Comment:
Unused method argument - `block`. If it's necessary, use `_` or `_block` as an argument name to indicate that it won't be used.